src/maxdiffusion/max_utils.py

```diff
@@ -342,98 +358,69 @@ def setup_initial_state(model, tx, config, mesh, model_params, unboxed_abstract_
     state: the initialized train state
     state_mesh_annotations: the mesh annotations for the train state
   """
-
+  max_logging.log(f"setup_initial_state for {checkpoint_item}")
   # Initialization
   state = None
+  unboxed_abstract_state, _, state_mesh_shardings = get_abstract_state(
+    model, tx, config, mesh, weights_init_fn, training)
   with nn_partitioning.axis_rules(config.logical_axis_rules):
     if checkpoint_manager:
-      state, raw_params = checkpointing.load_state_if_possible(checkpoint_manager,
-                                                  config.load_parameters_path,
-                                                  config.load_from_other_directory,
-                                                  config.load_from_other_directory_step,
-                                                  unboxed_abstract_state,
-                                                  mesh,
-                                                  state_mesh_annotations)
-
-    state_mesh_shardings = jax.tree_util.tree_map(
-        lambda p: jax.sharding.NamedSharding(mesh, p), state_mesh_annotations)
-  if not state:
-    init_train_state_partial = functools.partial(init_train_state, model=model, tx=tx, training=training)
-
-    sharding = PositionalSharding(mesh.devices).replicate()
-    partial_device_put_replicated = functools.partial(device_put_replicated, sharding=sharding)
-    model_params = jax.tree_util.tree_map(partial_device_put_replicated, model_params)
-
-    with jax.transfer_guard("disallow"):
+      state = checkpointing_utils.load_state_if_possible(checkpoint_manager,
+                                                  unboxed_abstract_state, checkpoint_item)
+      if state:
+        state = state[checkpoint_item]
+    if not state:
+      max_logging.log(f"Could not find {checkpoint_item} in orbax, creating state...")
+      # for DDP needs replication before jit state.
+      if config.ici_data_parallelism == -1 or config.dcn_data_parallelism == -1:
```

Comment:
Is it  safe to add ici_data_parallelism == np.prod(all_devices_axis)?  we might do fsdp but still pass ici_data_parallelism  = -1